src/vector3.h

```diff
@@ -136,12 +132,12 @@ class alignas(sizeof(T)) vector3 {
 	}
 	vector3 Normalized() const
 	{
-		const T l = 1.0f / sqrt(x * x + y * y + z * z);
-		return vector3(x * l, y * l, z * l);
+		const T l = Length();
```

Comment:
On the basis that `1.0 / sqrt(...)` may compile into an `rsqrt` instruction with significantly lower precision than `sqrt`, I'm willing to accept the conversion to division rather than multiplication by the reciprocal.

However, please do not replace manually-inlined operations like this with calls to a separate function. These inlined operations are written such to avoid function call overhead when compiling in debug mode for math operations that are called millions of times per second.